DependencyInjection/MonologExtension.php

```diff
@@ -49,29 +51,19 @@ class MonologExtension extends Extension
 
     private function levelToMonologConst($level, ContainerBuilder $container)
     {
-        if (null === $level || is_numeric($level)) {
-            return $level;
-        }
-
-        if (defined('Monolog\Logger::'.strtoupper($level))) {
-            return constant('Monolog\Logger::' . strtoupper($level));
-        }
-
-        if ($container->hasParameter($level)) {
-            return $this->levelToMonologConst($container->getParameter($level), $container);
-        }
-
-        try {
-            $logLevel = $container->resolveEnvPlaceholders($level, true);
-        } catch (ParameterNotFoundException $notFoundException) {
-            throw new \InvalidArgumentException(sprintf('Could not match "%s" to a log level.', $level));
-        }
+        if (is_string($level) && str_contains($level, '%')) {
```

Comment:
Resolution can be done at build time if it does not contains a parameter or env var.